### PR TITLE
QUIC wire protocol replacement

### PR DESCRIPTION
**Pull request by tewinget**

### CODE REVIEW

File: llarp/link/connection.hpp [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    RouterContact remote_rc;
+    AddressInfo remote_addr_info; // RC may have many, this is the one in use for this connection
+
+    bool inbound; // one side of a connection will be responsible for some things, e.g. heartbeat
+    bool remote_is_relay;
+
+    public:
+
+      const RouterContact& RemoteRC() { return remote_rc; }
+      const RouterID& RemoteID() { return remote_id; }
```
Comment (majestrate) [inline llarp/link/connection.hpp]: will this `RouterID` not just be the identity pubkey from the router contact?

File: llarp/link/connection.hpp [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    RouterID remote_id;
+    RouterContact remote_rc;
+    AddressInfo remote_addr_info; // RC may have many, this is the one in use for this connection
+
+    bool inbound; // one side of a connection will be responsible for some things, e.g. heartbeat
+    bool remote_is_relay;
+
+    public:
+
+      const RouterContact& RemoteRC() { return remote_rc; }
```
Comment (majestrate) [inline llarp/link/connection.hpp]: lacks const correctness.

File: llarp/link/connection.hpp [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+namespace llarp::link
+{
+
+  class Connection
+  {
+    std::shared_ptr<oxen::quic::connection_interface> conn;
+
+    RouterID remote_id;
+    RouterContact remote_rc;
+    AddressInfo remote_addr_info; // RC may have many, this is the one in use for this connection
```
Comment (majestrate) [inline llarp/link/connection.hpp]: could the `remote_addr_info` member be a const l value reference instead?

File: llarp/quic/endpoint.cpp [hunk 4]
```diff
@@ -130,7 +130,7 @@ namespace llarp::quic
   void
   Endpoint::handle_conn_packet(Connection& conn, const Packet& p)
   {
-    if (ngtcp2_conn_is_in_closing_period(conn))
```
Comment (majestrate) [inline llarp/quic/endpoint.cpp:133]: should this be cherry-picked into its own commit?

File: llarp/link/connection.hpp (hunk 1 above)
Comment (dr7ana) [reply llarp/link/connection.hpp]: resolving these on the WIP pr, messaged you offline

**Pull request closed without merging**